tests/src/testcommon.hh

```diff
@@ -224,49 +230,24 @@ template <typename NonLinearOperator, typename FiniteElement>
                                 .addResultRequest(ResultType::linearStress);
 
-  ResultTypeMap<double> result;
+  ResultTypeMap result;
   auto gridView        = fe.localView().globalBasis().gridView();
-  using GridView       = decltype(gridView);
   auto scalarBasis     = makeConstSharedBasis(gridView, lagrangeDG<1>());
   auto localScalarView = scalarBasis->localView();
   std::vector<Dune::FieldVector<double, 3>> stressVector(scalarBasis->size());
 
-  using LinearElasticElement
-      = Ikarus::LinearElastic<Ikarus::Basis<std::remove_cvref_t<decltype(fe.localView().globalBasis().preBasis())>>>;
-  using EASElement = Ikarus::EnhancedAssumedStrains<
-      Ikarus::LinearElastic<Ikarus::Basis<std::remove_cvref_t<decltype(fe.localView().globalBasis().preBasis())>>>>;
-
-  if constexpr (std::is_same_v<LinearElasticElement, FiniteElement> or std::is_same_v<EASElement, FiniteElement>) {
-    auto resultRequirements2 = Ikarus::ResultRequirements<>()
-                                   .insertGlobalSolution(Ikarus::FESolutions::displacement, displacement)
-                                   .addResultRequest(ResultType::PK2Stress);
-    Dune::Vtk::VtkWriter<GridView> vtkWriter(gridView);
-    std::vector<FiniteElement> fes;
-    fes.push_back(fe);
-    auto resultFunction
-        = std::make_shared<ResultFunction<std::remove_cvref_t<FiniteElement>>>(&fes, resultRequirements2);
-    try {
-      vtkWriter.addPointData(Dune::Vtk::Function<GridView>(resultFunction));
-      t.check(false) << "resultFunction5 should have failed for requesting PK2Stress here";
-    } catch (const Dune::NotImplemented&) {
-    }
-  } else
-    std::cout << "Result requirement check is skipped as " << Dune::className<FiniteElement>()
-              << " is not equivalent to " << Dune::className<LinearElasticElement>() << " or "
-              << Dune::className<EASElement>() << std::endl;
-
   auto ele = elements(gridView).begin();
   localScalarView.bind(*ele);
   const auto& fe2              = localScalarView.tree().finiteElement();
   const auto& referenceElement = Dune::ReferenceElements<double, gridDim>::general(ele->type());
-  for (auto c = 0UL; c < fe2.size(); ++c) {
+  for (const auto c : std::views::iota(0u, fe2.size())) {
     const auto fineKey                        = fe2.localCoefficients().localKey(c);
     const auto nodalPositionInChildCoordinate = referenceElement.position(fineKey.subEntity(), fineKey.codim());
     fe.calculateAt(resultRequirements, nodalPositionInChildCoordinate, result);
     Eigen::Vector3d computedResult = result.getResult(ResultType::linearStress);
     const auto nodeIndex           = localScalarView.index(localScalarView.tree().localIndex(c))[0];
     stressVector[nodeIndex]        = toDune(computedResult);
-    for (auto voigtIndex = 0UL; voigtIndex < 3; ++voigtIndex) {
+    for (const auto voigtIndex : std::views::iota(0u, 3u)) {
```

Comment:
Even though the test is only for 2D yet, i would write here `dim*(dim + 1) / 2` to directly support 3D